File: client/src/types/habit.ts
export const FREQUENCY_OPTIONS = ["Daily", "Weekly", "Monthly"] as const;
export type Frequency = typeof FREQUENCY_OPTIONS[number];

export type Habit = {
  id: string | number;
  name: string;
  category: string | null;
  frequency: Frequency;
  completedDays: number;
};

File: Models/Habit.cs
namespace HabitTracker.Models
{
    public class Habit
    {
        public int Id { get; set; }

        // Owner
        public string UserId { get; set; } = string.Empty;
        public string Name { get; set; } = string.Empty;
        public string? Category { get; set; }
        public string Frequency { get; set; } = string.Empty; // e.g., "Daily", "Weekly"
        public int CompletedDays { get; set; }
    }
}